src/main/java/net/neoforged/neoforge/client/model/data/ModelDataManager.java

```diff
@@ -91,14 +93,25 @@ public ModelData getAt(BlockPos pos) {
 
     /**
      * Snapshot the state of this manager for all sections in the volume specified by the given section coordinates.
-     *
-     * @throws IllegalArgumentException if this is a snapshot and the given region doesn't match the snapshot's region
+     * The snapshot will return {@link ModelData#EMPTY} for nonexistent keys.
      */
-    public ModelDataManager.Snapshot snapshotSectionRegion(int sectionMinX, int sectionMinY, int sectionMinZ, int sectionMaxX, int sectionMaxY, int sectionMaxZ) {
+    @Unmodifiable
```

Comment:
The `Unmodifiable` annotation only applies to collections and maps as per its documentation, so applying it to a function doesn't make sense.